File: social-service/src/main/java/com/eirsteir/coffeewithme/social/dto/FriendshipDto.java
package com.eirsteir.coffeewithme.social.dto;

import com.eirsteir.coffeewithme.commons.dto.UserDetailsDto;
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipStatus;
import lombok.*;
import lombok.experimental.Accessors;

@Getter
@Setter
@ToString
@Builder
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
public class FriendshipDto {

    private UserDetailsDto requester;
    private UserDetailsDto addressee;
    private FriendshipStatus status;

}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/dto/FriendshipDto.kt
package com.eirsteir.coffeewithme.social.dto

import com.eirsteir.coffeewithme.commons.dto.UserDetailsDto
import com.eirsteir.coffeewithme.social.domain.friendship.FriendshipStatus
import lombok.Builder
import lombok.Getter
import lombok.Setter
import lombok.experimental.Accessors


data class FriendshipDto(
    val requester: UserDetailsDto,
    val addressee: UserDetailsDto,
    val status: FriendshipStatus
)